src/kong-adapter/src/kong/utils.ts

```diff
@@ -298,6 +298,44 @@ function kongGet(url: string, callback: Callback<any>) {
     kongAction('GET', url, null, 200, callback);
 };
 
+function kongPagingGet(url: string, callback: Callback<any>) {
+    let pagingUrl;
+    const size = 1000;
+    if (url.indexOf('?') > 0)
+        pagingUrl = `${url}&size=${size}`;
+    else
+        pagingUrl = `${url}?size=${size}`;
+    console.log(pagingUrl);
+    const dataArray = [];
+    let finished = false;
+    async.until(function () {
+        return finished;
+    }, function (callback) {
+        kongGet(pagingUrl, function (err, result) {
+            if (err)
+                return callback(err);
+            if (!result.data)
+                return callback(new Error(`kongGet(${pagingUrl}) did not receive "data" property`));
+            for (let d of result.data)
+                dataArray.push(d);
```

Comment:
I wonder if typescript has this as js, `dataArray.push.apply(d)`, instead of this tight loop